### PR TITLE
Issa/39284/Redirecting to chosen trading platform

### PR DESCRIPTION
**Pull request by issajeanmarie**
Changes:

Redirecting user to the chosen/selected trading platform

- Redirects to mt5
- Redirects to dbot
- Redirects to derivx
- Redirects to Dtrader otherwise

## Type of change

-   [x] Bug fix
-   [x] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by suthesan-deriv — APPROVED** [review]

File: src/common/login.js [hunk 1]
```diff
@@ -26,9 +27,11 @@ const Login = (() => {
             ? `&affiliate_token=${affiliate_tracking}`
             : ''
 
+        const sub_url = redirectToTradingPlatform()
+
         return server_url && /qa/.test(server_url)
             ? `https://${server_url}/oauth2/authorize?app_id=${getAppId()}&l=${language}&brand=${brand_name.toLowerCase()}${affiliate_token_link}${cookies_link}`
-            : `${oauth_url}/oauth2/authorize?app_id=${deriv_app_id}&l=${language}&brand=${brand_name.toLowerCase()}${affiliate_token_link}${cookies_link}`
+            : `${oauth_url}/oauth2/authorize?app_id=${deriv_app_id}&l=${language}&brand=${brand_name.toLowerCase()}${affiliate_token_link}${cookies_link}&platform=${sub_url}`
```
Comment (sean-binary) [inline src/common/login.js:34]: We should also pass the platform query to the first condition as well.

File: src/common/utility.js [hunk 2]
```diff
@@ -244,3 +244,31 @@ export const redirectOpenLiveChatBox = (is_redirect) => {
...
+// Function which returns sub path to the specific trading platform
+export const redirectToTradingPlatform = () => {
+    const url_location = window.location.href
+
+    const dmt5_url = 'mt5'
+    const dbot_url = 'bot'
+    const derivx_url = 'derivx'
+
+    const isUserAt = () => {
+        const checkURL = (keyURL) => url_location.indexOf(keyURL) > -1
```
Comment (sean-binary) [inline src/common/utility.js:257]:
```suggestion
        const checkURL = (platform) => window.location.pathname.includes(platform)
```

Comment (sean-binary) [reply src/common/utility.js:257]: Doing something like this would be a clearer check

**Comment by issajeanmarie** [comment]
@sean-binary  and @suthesan-deriv can review the PR again (added requested changes) and let me know if I can update the card

**Review by alamin-deriv — APPROVED** [review]

File: src/common/utility.js [hunk 3]
```diff
@@ -244,3 +244,29 @@ export const redirectOpenLiveChatBox = (is_redirect) => {
         navigate(live_chat_redirection_link, { replace: true })
     }
 }
+
+// Function which returns sub path to the specific trading platform
+export const redirectToTradingPlatform = () => {
+    const dmt5_url = 'mt5'
+    const dbot_url = 'bot'
+    const derivx_url = 'derivx'
```
Comment (sean-binary) [inline src/common/utility.js:252]: Can we store this in an array/object instead?

File: src/common/utility.js [hunk 4]
```diff
@@ -244,3 +244,29 @@ export const redirectOpenLiveChatBox = (is_redirect) => {
...
+        if (isUserAt().dmt5) {
+            return dmt5_url
+        } else if (isUserAt().dbot) {
+            return dbot_url
+        } else if (isUserAt().derivx) {
+            return derivx_url
+        } else {
+            return ''
+        }
+    }
```
Comment (sean-binary) [inline src/common/utility.js:269]: Can we loop through the object/array and return its entries so that it's less repetitive instead?

Comment (issajeanmarie) [reply src/common/utility.js:269]: Yeah, this is how I refactored the codes:

```javascript
const supported_platforms = ['mt5', 'bot', 'derivx']
export const redirectToTradingPlatform = () =>
    supported_platforms.filter(
        (platform, index) =>
            window.location.pathname.includes(platform) && supported_platforms[index],
    )
```

Let me know if it's okay!

File: src/common/utility.js [hunk 5]
```diff
@@ -244,3 +244,11 @@ export const redirectOpenLiveChatBox = (is_redirect) => {
...
     }
 }
+
+// Function which returns sub path to the specific trading platform
+const supported_platforms = ['mt5', 'bot', 'derivx']
+export const redirectToTradingPlatform = () =>
+    supported_platforms.filter(
+        (platform, index) =>
+            window.location.pathname.includes(platform) && supported_platforms[index],
+    )
```
Comment (sean-binary) [inline src/common/utility.js:254]:
```suggestion
// Function which returns sub path to the specific trading platform
const supported_platforms = ['mt5', 'bot', 'derivx']
export const redirectToTradingPlatform = () =>
    supported_platforms.filter(
        (platform) =>
            window.location.pathname.includes(platform) && platform,
    )
```

Comment (sean-binary) [reply src/common/utility.js:254]: LGTM 👍  But in this case, you could just return the platform that is being looped through.

Comment (issajeanmarie) [reply src/common/utility.js]: Yeah, that's correct!

**Review by sean-binary — APPROVED** [review]
👍

**Pull request merged**